Add snap

florcabral opened the pull request:
Hello Space ROS team,

I would like to contribute [a snap](https://snapcraft.io/about) for this process_sarif project. If you’re not familiar with them, snaps are packages of an app and its dependencies for desktop, cloud and IoT. They are easy to install, secure, cross‐platform and self-contained.

A number of open source code security tools are already available to install as snaps (see [ikos](https://snapcraft.io/ikos), [cppcheck](https://snapcraft.io/cppcheckgui), [flawfinder](https://snapcraft.io/flawfinder)), but none other for SARIF file processing, which I think can make this an interesting contribution.

The snap is now [available on the snap store](https://snapcraft.io/process-sarif) (unlisted, so it cannot be found via the search engine). Give it a shot with `snap install process-sarif`. Currently, each of the 4 tools described in the main repo (sarif_load_test, conformance, duplicates, visualize) can be run as `process-sarif.<tool_name>` (ie: `process-sarif.sarif-load-test`) once the snap is installed. Other tools can easily be added to this snap as well. 

Please go ahead, give it a try and let me know what you think. If you would be interested in merging this and maintaining the snap, I can transfer the ownership to you on the snap store.

Please let me know if I can help answer any questions.

Cheers,
Florencia.

florcabral committed:
add snap section on README, add snapcraft.yaml

florcabral committed:
add snap link

florcabral commented:
Hello Space ROS team,

I would like to contribute [a snap](https://snapcraft.io/about) for this process_sarif project. If you’re not familiar with them, snaps are packages of an app and its dependencies for desktop, cloud and IoT. They are easy to install, secure, cross‐platform and self-contained.

A number of open source code security tools are already available to install as snaps (see [ikos](https://snapcraft.io/ikos), [cppcheck](https://snapcraft.io/cppcheckgui), [flawfinder](https://snapcraft.io/flawfinder)), but none other for SARIF file processing, which I think can make this an interesting contribution.

The snap is now [available on the snap store](https://snapcraft.io/process-sarif) (unlisted, so it cannot be found via the search engine). Give it a shot with `snap install process-sarif`. Currently, each of the 4 tools described in the main repo (sarif_load_test, conformance, duplicates, visualize) can be run as `process-sarif.<tool_name>` (ie: `process-sarif.sarif-load-test`) once the snap is installed. Other tools can easily be added to this snap as well. 

Please go ahead, give it a try and let me know what you think. If you would be interested in merging this and maintaining the snap, I can transfer the ownership to you on the snap store.

Please let me know if I can help answer any questions.

Cheers,
Florencia.

mjeronimo commented:
@nuclearsandwich What do you think of enabling snap for the SARIF processing tools?

mjeronimo commented:
@florcabral I tried out the snap and it works fine. I'm willing to merge the changes, but can't take on the maintenance as I'm no longer officially on Space ROS since the acquisition of OSRC by Intrinsic (I am spending some personal time to help in the transition).

EzraBrooks commented on README.md:
```diff
@@ -30,4 +30,12 @@ These are files that aren't run directly, but can be imported by your own script
\ No newline at end of file
+    - Many Cobra files cannot be loaded due to invalid JSON, or non-conformance to the SARIF spec.
+
+## 4. Snap
+These tools are also available to install from the [Snap Store](https://snapcraft.io/about) as a [snap](https://snapcraft.io/process-sarif).
+### Install: 
+`snap install process-sarif`
+### Usage
+Navigate to a directory containing the SARIF file(s) you wish to analyze. Run any of the available commands (sarif-load-test, conformance, duplicates, visualize), as `process-sarif.<app_name>` (ie: process-sarif.sarif-load-test)
```
As discussed in the meeting earlier, I would love some wording here to the effect of this being a non-default method of installation that is not maintained by core Space ROS contributors but is instead maintained by Canonical.

florcabral replied:
@EzraBrooks Thanks for the review, and for the discussion. As agreed, I have adjusted the description in the README to make it clear that the snap is only a complementary packaging option maintained by Canonical for the time being. We will measure user adoption to decide what will happen to it in the future.

florcabral committed:
add github action

adds a Github action that will be triggered on changes on the main branch, a new tag, or when a pull request is opened against main. The action builds, tests, and publishes the snap.

florcabral committed:
add github action

florcabral committed:
Merge branch 'add-snap' of https://github.com/florcabral/process_sarif into add-snap

florcabral committed:
update snap description on README
